samples/display/lvgl/boards/stm32f429i_disc1.conf

```diff
@@ -0,0 +1,4 @@
+CONFIG_LVGL=y
```

Comment:
This config overlay is not necessary if the display is a part of the board, better to define/configure it on the boards level in the Kconfig.defconfig file, see https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/mimxrt1050_evk/Kconfig.defconfig#L49

Reply:
Good point, actually it's the best approach :+1:

Reply:
Good point. So should the lvgl example not be using overlays at all then? I used the approach used in the lvgl example for other boards.

Reply:
@jfischer-no but then another things comes to mind. Why should board config care about LVGL? That is backwards. Board config should provide a service for LVGL to run on (ie framebuffer). LVGL should be configured separately and only care about service being there through the "DISPLAY" label. What if there is another gui toolkit used? Would we then have to add a bunch of other ifdefs into the board file to test for it? It actually seems more reasonable to enable LVGL along with SPI as @gmarull pointed out at application level inside the particular board overlay.

Reply:
I assume IMX boards do that because they are supposed to support LVGL only, but yes, if multiple frameworks are to be used it’s not the right choice as it’s a dependency specified in the opposite direction.

Reply:
So summarizing: I still think overlay (same as reel board) is the best choice in the end.

Reply:
Actually line 1 should be removed and the remaining  configs can be moved to the board level Kconfig.defconfig, as such they will provide the _default_ values if `LVGL`.

Reply:
Now both solutions Kconfig.defconfig and config overlays are bad. But Kconfig.defconfig does not litter the samples and is only found in one place, so easier to maintain. Actually we need display-controller.yaml binding which is included by each controller and contains the generic properties. LVGL should derive the properties from device tree and probably supports multiple displays.